### PR TITLE
Extract out logic for creating custom buttons

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# What

This is the second PR in a series to simplify `CallControls` to the point where it need not be shared across composites.

This PR extracts out the logic for creating custom buttons. It also simplifies the conditional compilation logic quite a bit by declaring the new types unconditionally, but exporting them from the package conditionally.

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 1]
```diff
@@ -132,50 +111,31 @@ export const CallControls = (props: CallControlsProps): JSX.Element => {
...
               displayType={options?.displayType}
               increaseFlyoutItemSize={props.increaseFlyoutItemSize}
               styles={props.commonButtonStyles}
             />
           )}
-          {
-            /* @conditional-compile-remove-from(stable): custom button injection */
-            <FilteredCustomButtons customButtonProps={customButtonProps} placement={'afterOptionsButton'} />
-          }
+          {customButtons['afterOptionsButton']}
```
Comment (prprabhu-ms) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:122]: Not for this PR: This should be named `afterDevicesButton`.

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 2]
```diff
@@ -132,50 +111,31 @@ export const CallControls = (props: CallControlsProps): JSX.Element => {
...
-  return (
-    <>
-      {props.customButtonProps
-        .filter((buttonProps) => buttonProps.placement === props.placement)
-        .map((buttonProps, i) => (
-          <ControlBarButton {...buttonProps} key={`${buttonProps.placement}_${i}`} />
-        ))}
-    </>
-  );
+const onFetchCustomButtonPropsTrampoline = (
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:134]: `onFetchCustomButtonPropsTrampoline` 
What's the story behind this name?

Comment (dmceachernmsft) [reply packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:134]: I mean I like a good trampoline but its also got me wondering

**Review by anjulgarg — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx (hunk 2 above)
Comment (prprabhu-ms) [reply packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:134]: Heh, old habit. `trampoline` is a term used to refer to helper functions for FFI bindings.

image

from https://www.gnu.org/software/mit-scheme/documentation/stable/mit-scheme-ffi.pdf

Comment (prprabhu-ms) [reply packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:134]: btw, 100% onboard with a more meaningful name for these kinds of functions for conditional compilation.

File: packages/react-composites/src/composites/CallComposite/types/CallControlOptions.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-/* @conditional-compile-remove-from(stable): custom button injection */
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/types/CallControlOptions.ts:4]: Are we going to say we wanna check this into stable directly?

Comment (prprabhu-ms) [reply packages/react-composites/src/composites/CallComposite/types/CallControlOptions.ts:4]: This is defined unconditionally, but only exported for beta.

We've found that this pattern makes for cleaner code and smaller diff between the build flavors -- define the types unconditionally internally, but only export in the beta flavor.

**Review by alkwa-msft — APPROVED** [review]

**Pull request merged**